gpjax/decision_making/decision_maker.py

```diff
@@ -222,28 +240,64 @@ class UtilityDrivenDecisionMaker(AbstractDecisionMaker):
     utility_function_builder: AbstractUtilityFunctionBuilder
     utility_maximizer: AbstractUtilityMaximizer
 
-    def ask(self, key: KeyArray) -> Float[Array, "1 D"]:
+    def __post_init__(self):
+        super().__post_init__()
+        if self.batch_size > 1 and not isinstance(
+            self.utility_function_builder, ThompsonSampling
+        ):
+            raise NotImplementedError(
+                "Batch size > 1 currently only supported for Thompson sampling."
+            )
+
+    def ask(self, key: KeyArray) -> Float[Array, "B D"]:
         """
-        Get updated utility function and return the point which maximises it. This
-        method also stores the utility function in
-        `self.current_utility_function` so that it can be accessed after the ask
+        Get updated utility function(s) and return the point(s) which maximises it/them. This
+        method also stores the utility function(s) in
+        `self.current_utility_functions` so that they can be accessed after the ask
         function has been called. This is useful for non-deterministic utility
-        functions, which will differ between calls to `ask` due to the splitting of
+        functions, which may differ between calls to `ask` due to the splitting of
         `self.key`.
 
+        Note that in general `SinglePointUtilityFunction`s are only capable of
+        generating one point to be queried at each iteration of the decision making loop
+        (i.e. `self.batch_size` must be 1). However, Thompson sampling can be used in a
+        batched setting by drawing a batch of different samples from the GP posterior.
+        This is done by calling `build_utility_function` with different keys
+        sequentilly, and optimising each of these individual samples in sequence in
+        order to obtain `self.batch_size` points to query next.
+
         Args:
             key (KeyArray): JAX PRNG key for controlling random state.
 
         Returns:
-            Float[Array, "1 D"]: Point to be queried next.
+            Float[Array, "B D"]: Point(s) to be queried next.
         """
-        self.current_utility_function = (
-            self.utility_function_builder.build_utility_function(
-                self.posteriors, self.datasets, key
+        self.current_utility_functions = []
+        maximizers = []
+        if isinstance(self.utility_function_builder, ThompsonSampling) or (
+            (not isinstance(self.utility_function_builder, ThompsonSampling))
+            and (self.batch_size == 1)
```

Comment:
does this still not work with batch_size=1? This logic is a little tricky to understand

Reply:
It's restricting batch size to be 1 for non-TS acquisition functions if someone happens to implement one for now.

Reply:
got it. Maybe you can make the logic clearer so people like me dont get confused?